lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -51,6 +51,27 @@ void SubstraitDialect::initialize() {
 namespace mlir {
 namespace substrait {
 
+LogicalResult
+CrossOp::inferReturnTypes(MLIRContext *context, std::optional<Location> loc,
+                          ValueRange operands, DictionaryAttr attributes,
+                          OpaqueProperties properties, RegionRange regions,
+                          llvm::SmallVectorImpl<Type> &inferredReturnTypes) {
+  Value leftInput = operands[0];
+  Value rightInput = operands[0];
+
+  TypeRange leftFieldTypes = leftInput.getType().cast<TupleType>().getTypes();
+  TypeRange rightFieldTypes = rightInput.getType().cast<TupleType>().getTypes();
+
+  llvm::SmallVector<mlir::Type> fieldTypes;
+  fieldTypes.append(leftFieldTypes.begin(), rightFieldTypes.end());
```

Comment:
Should this be leftFiledTypes.end() ? (I think ADT has some helpers to work with ranges)

Reply:
Good catch! I had only looked in the member functions of `SmallVector` and was surprised (annoyed?) that it wasn't there. I do see that there is a free function, and the newest version uses that. Still a bit surprised that it's only a free function...